logstage/logstage-sink-file/src/main/scala/com/github/pshirshov/izumi/FileSink.scala

```diff
@@ -0,0 +1,88 @@
+package com.github.pshirshov.izumi
+
+import java.util.concurrent.atomic.AtomicReference
+import com.github.pshirshov.izumi.models._
+import com.github.pshirshov.izumi.logstage.api.logger.RenderingPolicy
+import com.github.pshirshov.izumi.logstage.model.Log
+import com.github.pshirshov.izumi.logstage.model.logger.LogSink
+import com.github.pshirshov.izumi.logstage.sink.console.ConsoleSink
+import scala.util.{Failure, Success, Try}
+
+
+class FileSink(policy: RenderingPolicy, consoleSink: ConsoleSink, config: FileSinkConfig) extends LogSink {
+
+  private val sinkState = new AtomicReference[SinkState](SinkState(config = config))
+
+  def processCurrentFile(state: SinkState): Try[SinkState] = {
+    val curFileId = state.currentFileId
+    val updatedStatus = {
+      if (state.currentFileSize == FileSize.undefined) {
+        state.getFile(curFileId).flatMap {
+          file =>
+            file.getFileSize.map {
+              case size if size > 0 =>
+                state.config.rotation.filesLimit.map(l => {
+                  (curFileId + 1) to l map {
+                    id => state.removeFile(id)
+                  }
+                })
+                state.copy(currentFileSize = size)
+              case size =>
+                state.copy(currentFileSize = size)
+            }
+        }
+      } else {
+        Success(state)
+      }
+    }
+
+    updatedStatus.flatMap {
```

Comment:
I guess just `.map` would be enough here. Also consider moving the logic into a method and invoke it like `updatedStatus.map(computeFileId)`